### PR TITLE
Group detail user aware

### PR DESCRIPTION
**Pull request by nicknapoli82**
Adding client application user awareness for group context. 
It knows if the user is logged in or not, and allows a user to join a group or leave a group.

NOT COMPLETE. Just helper code to allow everyone else work on their own slice of the application

### CODE REVIEW

File: client/components/Group/GroupDetail.js [hunk 1]
```diff
@@ -10,26 +11,59 @@ class GroupDetail extends Component {
...
+  }
 
   handleItemClick = (e, { name }) => {
     this.setState({ activeItem: name, context: name });
     this.props.getDetailGroup(this.props.match.params.groupId, name);
   };
 
+  async joinGroup(match) {
+    try {
+      const result = axios.post('/api/groups/addself', {groupId: match.params.groupId});
```
Comment (justin-cook-developer) [inline client/components/Group/GroupDetail.js]: I think you need to await here.

File: client/components/Group/GroupDetail.js [hunk 2]
```diff
@@ -10,26 +11,59 @@ class GroupDetail extends Component {
...
+    }
+    catch(e) {
+      console.log('Failed to add user to group?');
+    }
+  }
+
+  leaveGroup(match) {
+    console.log(match);
+    try {
+      const result = axios.delete('/api/groups/removeself', {data: {groupId: match.params.groupId}});
```
Comment (justin-cook-developer) [inline client/components/Group/GroupDetail.js]: await?

File: client/components/Group/GroupDetail.js [hunk 3]
```diff
@@ -10,26 +11,59 @@ class GroupDetail extends Component {
...
+  }
+
   componentDidMount() {
     this.props.getDetailGroup(this.props.match.params.groupId, this.state.context);
     this.setState({ groupId: this.props.match.params.groupId});
   }
 
   compontentDidUpdate() {
     const { context } = this.state;
+    if(this.props.members.length && this.props.user.id !== undefined) {
```
Comment (justin-cook-developer) [inline client/components/Group/GroupDetail.js]: Does this put you in an infinite loop? I think you might want to add a conditional to check if isMember is already true. didUpdate has the signature (prevProps, prevState)

**Review by justin-cook-developer — CHANGES_REQUESTED** [review]

**Comment by nicknapoli82** [comment]
Alright. I made some fair changes to how the data works through this slice of the app now. 
Moved some of the Component state out of the GroupDetail component and into the groups redux store. Pretty much solves the problems I created earlier.

Let me know if you see anything that looks wrong. 

Still not done, but this is a better working model now IMO

**Review by justin-cook-developer — APPROVED** [review]

**Pull request merged**